gc/base/MemorySubSpaceSemiSpace.cpp

```diff
@@ -961,6 +961,9 @@ MM_MemorySubSpaceSemiSpace::checkSubSpaceMemoryPostCollectResize(MM_EnvironmentB
 				_expansionSize = MM_Math::roundToCeiling(extensions->heapAlignment, (uintptr_t)(getCurrentSize() * adjustedExpansionFactor));
 				_expansionSize = MM_Math::roundToCeiling(2 * regionSize, _expansionSize);
 
+				/* Adjust within -XsoftMx limit */
+				_expansionSize = adjustExpansionWithinSoftMax(env, _expansionSize, 0);
```

Comment:
This is an ok approach, but may cause behavior not exactly that some people would expect.

When SoftMx changes, I think I'd rather change SemiSpace::_maximumSize  and SubSpaceGenerational::_maximumSizeNew   to be 1/4 of that of that softMx value (or better, have separate fields that are softMx specific, like _maximumSoftSize). Then no changes here would be needed here. Eventually in MM_MemorySubSpace::expand() we additionally cap with maxExpansionInSpace() that would account for that adjusted or new softMx limit.

As proposed here, we could go beyond 1/4 of softMx, till 1/4 of mx (what potentially is much larger), but till total heap size is within softMx.

Please, have a look if such an approach would is feasible (there might be some weird consequences, that I'm overlooking @dmitripivkine)